pyfixest/did/saturated_twfe.py

```diff
@@ -411,35 +431,60 @@ def compute_period_weights(
         Column name of treatment indicator (0/1).
     include_grid : bool, default True
         If True, returns a full (cohort x period) grid with zero-filled weights.
+    weights : Optional[str], default None
+        If provided, the name of the column in `data` that contains weights.
+    use_weights : bool, default True
+        If True, uses the analytic weights provided in the `weights` column for aggregation.
+        If False, uses simple counts.
 
     Returns
     -------
     pd.DataFrame
         Columns [cohort, period, weight]. If `include_grid`, every
         combination appears (with weight=0 where not defined).
     """
-    df = data[[cohort, period, treatment]].copy()
+    columns = [cohort, period, treatment]
+    if weights is not None:
+        columns.append(weights)
+
+    df = data[columns].copy()
     ever_treated = df.loc[df[treatment] == 1, cohort].unique()
 
     # post-treatment cells (l > 0)
-    post = (
-        df[df[treatment] == 1]
-        .groupby([cohort, period])
-        .size()
-        .reset_index(name="n_grel")
-    )
+    if weights is not None and use_weights:
+        post = (
+            df[df[treatment] == 1]
+            .groupby([cohort, period])[weights]
+            .sum()
+            .reset_index(name="n_grel")
+        )
+    else:
+        post = (
+            df[df[treatment] == 1]
+            .groupby([cohort, period])
+            .size()
+            .reset_index(name="n_grel")
+        )
     post = post[post[period] >= 0]
     denom_post = post.groupby(period)["n_grel"].sum().reset_index(name="n_rel")
     post = post.merge(denom_post, on=period)
     post["weight"] = post["n_grel"] / post["n_rel"]
 
     # pre-treatment cells (l < 0)
-    pre = (
-        df[(df[treatment] == 0) & (df[cohort].isin(ever_treated))]
-        .groupby([cohort, period])
-        .size()
-        .reset_index(name="n_grel")
-    )
+    if weights is not None and use_weights:
```

Comment:
For some reason, no tests seem to be triggered for the aggregation method with weights, could you please add them to the [saturated_twfe tests](https://github.com/py-econometrics/pyfixest/blob/bc423748c50cf3f8297254e843813f787f1d3a98/tests/test_did.py#L338)?

Reply:
Just did a spot check and agree that it is likely an error in the CI / test setup. 

When running 

```python
import pyfixest as pf
import numpy as np
import pandas as pd

rng = np.random.default_rng(1243)
df_het = pd.read_csv("pyfixest/did/data/df_het.csv")
df_het["weights"] = rng.uniform(0, 10, size=len(df_het))
df_het["X"] = rng.normal(size=len(df_het))
print(df_het.head())


saturated_py = pf.event_study(
    data=df_het,
    yname="dep_var",
    idname="unit",
    tname="year",
    gname="g",
    estimator="saturated",
    cluster="unit",
    weights="weights",
)

saturated_py.aggregate()
```
which is more or less what happens in the added tests, the lines flagged as non-covered are actually reached.

Reply:
Yes, sorry? My mistake - this test should run under the test-against-fixest ci run.

Reply:
Fairly certain now that it is an upload error to codecov. I've tried to change the CI, let's see if that solves it =)

Reply:
Looks like it's fixed! 🎉